### PR TITLE
Bug 1162578 - Use etag caching for performance data

### PR DESCRIPTION
**Pull request by wlach**
It can get quite big so this is helpful

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/524)

### CODE REVIEW

**Comment by maurodoglio** [comment]
Good idea! A couple of questions:
- what's the tipical size of the response?
- what's the expiration policy? You are probably missing a list_cache_key_func function, like the one used [here](https://github.com/mozilla/treeherder/blob/54727c2994d9f937bd6297dadd000580bfcfc891/treeherder/webapp/api/refdata.py#L48).
If case you just wanted to enable the etag calculation, you can use EtagMixin or one of its subclasses

**Comment by wlach** [comment]
Oops, I actually don't think we want to do this. :) I just wanted the etag behaviour, not the cache behaviour. I'll redo it.

**Comment by wlach** [comment]
New PR. Let's just use etags for the series summary for now.

**Comment by wlach** [comment]
Ok that didn't work because it seems as if by default etag caching doesn't actually look at what response would have been sent, we have to calculate it ourselves. Updated patch with that enabled. As a bonus, this will give the correct behaviour if the series summary changed but the contents didn't (which is very common)

File: treeherder/webapp/api/performance_data.py [hunk 1]
```diff
@@ -15,7 +19,17 @@ class PerformanceDataViewSet(viewsets.ViewSet):
     This view serves performance charts data
     """
 
+    def _calculate_etag(view_instance, view_method,
+                       request, args, kwargs):
+        if not kwargs.get('project') or request.QUERY_PARAMS.get('interval'):
```
Comment (maurodoglio) [inline treeherder/webapp/api/performance_data.py]: Can you make this condition more explicit with some parenthesis?
Looking at the code below I think you want both project and interval to be there, right?

**Comment by maurodoglio** [comment]
here we go :shipit:

**Pull request merged**